ouroboros-consensus/src/Ouroboros/Consensus/Protocol/PBFT.hs

```diff
@@ -208,9 +208,10 @@ instance (PBftCrypto c, Typeable c) => OuroborosTag (PBft c) where
 
       -- We always include slot number 0 in case there are no signers yet.
       let lastSlot = maximum . (SlotNo 0 :)
```

Comment:
Oh, I would factor out the pattern-match (now I see why you were objecting :slightly_smiling_face:):
```haskell
let lastSlotOfSigner = \case
      _ Seq.:|> s -> s
      _           -> SlotNo 0
    lastSlot =  maximum
             .  (SlotNo 0 :)
             $  lastSlotOfSigner
            <$> Map.elems chainState
```
(I don't haven't type-checked this)